BUG: problem building bits/timesize.h not found on Ubuntu 22.04

### Describe your issue.

While building SciPy, compilation terminates becauses `bits/timesize.h` is not found.

### Reproducing Code Example

```python
python dev.py build
```


### Error message

```shell
[801/1588] Compiling C++ object scipy/stats/_stats_pythran.cp...x86_64-linux-gnu.so.p/meson-generated_..__stats_pythran.cpp.o
FAILED: scipy/stats/_stats_pythran.cpython-310-x86_64-linux-gnu.so.p/meson-generated_..__stats_pythran.cpp.o 
/home/gert/.conda/envs/scipy-dev/bin/x86_64-conda-linux-gnu-c++ -Iscipy/stats/_stats_pythran.cpython-310-x86_64-linux-gnu.so.p -Iscipy/stats -I../scipy/stats -I../../../.conda/envs/scipy-dev/lib/python3.10/site-packages/pythran -I../../../.conda/envs/scipy-dev/lib/python3.10/site-packages/numpy/core/include -I/usr/include -I/home/gert/.conda/envs/scipy-dev/include/python3.10 -fvisibility=hidden -fvisibility-inlines-hidden -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c++14 -O2 -g -fvisibility-inlines-hidden -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /home/gert/.conda/envs/scipy-dev/include -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /home/gert/.conda/envs/scipy-dev/include -fPIC -DENABLE_PYTHON_MODULE -D__PYTHRAN__=3 -DPYTHRAN_BLAS_NONE -Wno-cpp -Wno-deprecated-declarations -Wno-unused-but-set-variable -Wno-unused-function -Wno-unused-variable -Wno-int-in-bool-context -MD -MQ scipy/stats/_stats_pythran.cpython-310-x86_64-linux-gnu.so.p/meson-generated_..__stats_pythran.cpp.o -MF scipy/stats/_stats_pythran.cpython-310-x86_64-linux-gnu.so.p/meson-generated_..__stats_pythran.cpp.o.d -o scipy/stats/_stats_pythran.cpython-310-x86_64-linux-gnu.so.p/meson-generated_..__stats_pythran.cpp.o -c scipy/stats/_stats_pythran.cpp
In file included from /usr/include/features.h:392,
                 from /home/gert/.conda/envs/scipy-dev/x86_64-conda-linux-gnu/include/c++/12.3.0/x86_64-conda-linux-gnu/bits/os_defines.h:39,
                 from /home/gert/.conda/envs/scipy-dev/x86_64-conda-linux-gnu/include/c++/12.3.0/x86_64-conda-linux-gnu/bits/c++config.h:655,
                 from /home/gert/.conda/envs/scipy-dev/x86_64-conda-linux-gnu/include/c++/12.3.0/type_traits:38,
                 from ../../../.conda/envs/scipy-dev/lib/python3.10/site-packages/pythran/pythonic/include/types/assignable.hpp:4,
                 from ../../../.conda/envs/scipy-dev/lib/python3.10/site-packages/pythran/pythonic/types/assignable.hpp:4,
                 from ../../../.conda/envs/scipy-dev/lib/python3.10/site-packages/pythran/pythonic/core.hpp:41,
                 from scipy/stats/_stats_pythran.cpp:1:
/usr/include/features-time64.h:21:10: fatal error: bits/timesize.h: No such file or directory
   21 | #include <bits/timesize.h>
      |          ^~~~~~~~~~~~~~~~~
compilation terminated.
```


### SciPy/NumPy/Python version and system information

```shell
sys.version_info(major=3, minor=10, micro=12, releaselevel='final', serial=0)

SciPy version information not available because it cannot be built.
```

--- a/meson.build
+++ b/meson.build
@@ -183,11 +183,7 @@ tempita = find_program('scipy/_build_utils/tempita.py')
 
 use_pythran = get_option('use-pythran')
 if use_pythran
-  pythran = find_program('pythran', native: true, version: '>=0.14.0')
-  # xsimd is unvendored from pythran by conda-forge, and due to a compiler
-  # activation bug the default <prefix>/include/ may not be visible (see
-  # gh-15698). Hence look for xsimd explicitly.
-  xsimd_dep = dependency('xsimd', required: false)
+  pythran = find_program('pythran', native: true, version: '>=0.18.1')
 endif
 
 # Subprojects

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -20,7 +20,7 @@ requires = [
     "meson-python>=0.15.0",
     "Cython>=3.0.8",        # when updating version, also update check in meson.build
     "pybind11>=2.13.2",     # when updating version, also update check in scipy/meson.build
-    "pythran>=0.14.0",
+    "pythran>=0.18.1",      # when updating version, also update check in meson.build
 
     # numpy requirement for wheel builds for distribution on PyPI - building
     # against 2.x yields wheels that are also compatible with numpy 1.x at

--- a/scipy/meson.build
+++ b/scipy/meson.build
@@ -121,7 +121,6 @@ print(incdir)
   endif
   pythran_dep = declare_dependency(
     include_directories: incdir_pythran,
-    dependencies: xsimd_dep,
   )
 else
   pythran_dep = []

